kotlin/internal/jvm/impl.bzl

```diff
@@ -289,6 +290,12 @@ def kt_jvm_junit_test_impl(ctx):
     jvm_flags = []
     if hasattr(ctx.fragments.java, "default_jvm_opts"):
         jvm_flags = ctx.fragments.java.default_jvm_opts
+
+    java_toolchain = find_java_toolchain(ctx, ctx.attr._java_toolchain)
+
+    if java_toolchain and java_toolchain.java_runtime and java_toolchain.java_runtime.version > 18:
```

Comment:
I'd copy what Bazel does exactly so that we have some consistency. Can you add a comment as well linking out to their implementation for reference? https://github.com/bazelbuild/bazel/blob/6d5b084025a26f2f6d5041f7a9e8d302c590bc80/src/main/starlark/builtins_bzl/bazel/java/bazel_java_binary.bzl#L66-L67